src/plonk_gates.rs

```diff
@@ -1373,3 +1373,89 @@ impl<C: HaloCurve> WitnessGenerator<C::ScalarField> for ArithmeticGate<C> {
         result
     }
 }
+
+/// Test that a gate's constraints are within degree 8n, including the gate prefix filter.
+#[macro_export]
+macro_rules! test_gate_low_degree {
+    ($method:ident, $curve:ty, $gate:ty) => {
+        #[test]
+        #[ignore] // Too slow to run regularly.
+        fn $method() {
+            type C = $curve;
+            type SF = <C as $crate::curve::Curve>::ScalarField;
+
+            let n = 1024;
+            let fft_precomputation_n = $crate::fft::fft_precompute::<SF>(n);
+            let fft_precomputation_8n = $crate::fft::fft_precompute::<SF>(8 * n);
+
+            // Generate random constant and wire polynomials.
+            let mut constant_values_n: Vec<Vec<SF>> = vec![Vec::new(); $crate::plonk::NUM_CONSTANTS];
+            let mut wire_values_n: Vec<Vec<SF>> = vec![Vec::new(); $crate::plonk::NUM_WIRES];
+            for i in 0..n {
+                for points in constant_values_n.iter_mut() {
+                    points.push(<SF as $crate::field::Field>::rand())
+                }
+                for points in wire_values_n.iter_mut() {
+                    points.push(<SF as $crate::field::Field>::rand())
+                }
+            }
+
+            // Low-degree extend them to 8n values.
+            let constant_coeffs_n = $crate::plonk_util::values_to_coeffs(&constant_values_n, &fft_precomputation_n);
+            let constant_values_8n = $crate::plonk_util::coeffs_to_values_padded(&constant_coeffs_n, &fft_precomputation_8n);
+            let wire_coeffs_n = $crate::plonk_util::values_to_coeffs(&wire_values_n, &fft_precomputation_n);
+            let wire_values_8n = $crate::plonk_util::coeffs_to_values_padded(&wire_coeffs_n, &fft_precomputation_8n);
+
+            // Make sure each extended polynomial is still degree n.
+            for values_8n in &constant_values_8n {
+                assert_eq!($crate::plonk_util::polynomial_degree(values_8n, &fft_precomputation_8n), n - 1);
+            }
+            for values_8n in &wire_values_8n {
+                assert_eq!($crate::plonk_util::polynomial_degree(values_8n, &fft_precomputation_8n), n - 1);
+            }
```

Comment:
Shouldn't this be `assert!($crate::plonk_util::polynomial_degree(values_8n, &fft_precomputation_8n) <= n - 1);` instead? These polynomial should be of degree `n-1` with overwhelming probability though, so it shoudn't matter.

Reply:
Good point -- fixed.